src/main/java/com/valmiki/adhyapakservice/service/CourseServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.valmiki.adhyapakservice.service;
+
+import com.valmiki.adhyapakservice.dto.request.CourseRequest;
+import com.valmiki.adhyapakservice.dto.response.CourseResponse;
+import com.valmiki.adhyapakservice.entity.Course;
+import com.valmiki.adhyapakservice.exception.DataNotFoundException;
+import com.valmiki.adhyapakservice.repo.CourseRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CourseServiceImpl implements CourseService {
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Override
+    public List<CourseResponse> findAll() {
+        List<Course> courseList = courseRepository.findAll();
+        // TODO: Remove below code and use mapstruct for mapping
+        // TODO: And also use records for dtos
+        return courseList.stream()
+                .map(e -> CourseResponse.builder()
+                        .id(e.getId())
+                        .courseName(e.getCourseName())
+                        .subHeading(e.getSubHeading())
+                        .build()).toList();
+    }
+
+    @Override
+    public List<CourseResponse> findByID(int theId) {
```

Comment:
Why are we using list here, can there be multiple Course with same id

Reply:
fixed